libs/utils/utils.js

```diff
@@ -593,11 +606,76 @@ function decorateFooterPromo(config) {
   document.querySelector('main > div:last-of-type').insertAdjacentElement('afterend', section);
 }
 
-async function loadMartech(config) {
+export function loadIms(onReadyFn) {
+  if (window.adobeIMS) return;
+
+  const { locale, imsClientId, imsScope, env, onReady } = getConfig();
+  if (!imsClientId) return null;
+
+  window.adobeid = {
+    client_id: imsClientId,
+    scope: imsScope || 'AdobeID,openid,gnav',
+    locale: locale?.ietf?.replace('-', '_') || 'en_US',
+    autoValidateToken: true,
+    environment: env.ims,
+    useLocalStorage: false,
+    onReady: onReadyFn || onReady,
+  };
+  loadScript('https://auth.services.adobe.com/imslib/imslib.min.js');
+}
+
+async function loadMartech({ persEnabled = false, persManifests = [] } = {}) {
+  if (window.marketingtech?.adobe?.launch !== undefined) {
+    return true;
+  }
+
   const query = new URL(window.location.href).searchParams.get('martech');
-  if (query !== 'off' && getMetadata('martech') !== 'off') {
-    const { default: martech } = await import('../martech/martech.js');
-    martech(config, loadScript, getMetadata);
+  if (query === 'off' || getMetadata('martech') === 'off') {
+    return false;
+  }
+
+  window.targetGlobalSettings = { bodyHidingEnabled: false };
+  loadIms();
+
+  const { default: initMartech } = await import('../martech/martech.js');
+  await initMartech({
+    persEnabled,
+    persManifests,
+    utils: {
+      createTag, getConfig, getMetadata, loadLink, loadScript, updateConfig,
+    },
+  });
+
+  return true;
+}
+
+async function checkForPageMods() {
+  const persMd = getMetadata('personalization');
+  let persManifests = [];
```

Comment:
Do we have an early return opportunity here?

Reply:
Can't early return as we also need to check target (line 666)